File: StadiumEngine/StadiumEngine.WebUI/ClientApp/src/models/command/rates/AddTariffCommand.ts
import {PromoCodeDto, TariffDayIntervalDto} from "../../dto/rates/TariffDto";

export interface AddTariffCommand {
    name: string;
    description: string | null;
    isActive: boolean;
    dateStart: Date;
    dateEnd: Date | null;
    monday: boolean;
    tuesday: boolean;
    wednesday: boolean;
    thursday: boolean;
    friday: boolean;
    saturday: boolean;
    sunday: boolean;
    dayIntervals: TariffDayIntervalDto[],
    promoCodes: PromoCodeDto[]
}

File: Unfield/Unfield.Commands/Rates/Tariffs/AddTariffCommand.cs
using Mediator;
using Unfield.DTO.Rates.Tariffs;

namespace Unfield.Commands.Rates.Tariffs;

public sealed class AddTariffCommand : BaseCommand, IRequest<AddTariffDto>
{
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public bool IsActive { get; set; }
    public DateTime DateStart { get; set; }
    public DateTime? DateEnd { get; set; }
    public bool Monday { get; set; }
    public bool Tuesday { get; set; }
    public bool Wednesday { get; set; }
    public bool Thursday { get; set; }
    public bool Friday { get; set; }
    public bool Saturday { get; set; }
    public bool Sunday { get; set; }
    public List<TariffDayIntervalDto> DayIntervals { get; set; } = new();
    public List<PromoCodeDto> PromoCodes { get; set; } = new();
}